test/image/mocks/zzz-automargin-title-container.json

```diff
@@ -18,11 +18,14 @@
         "height": 300,
         "width": 400,
         "margin": {"t":0, "b": 0, "l": 0, "r": 0},
+        "xaxis": {"automargin": true, "title": {"text": "x-axis title", "standoff": 0}},
```

Comment:
@alexcjohnson - The bottom margin ends up being set to the correct size to account for both the plot title and the axis, but the axis title then still moves to overlap with the plot title when given enough margin space. Manually setting the `standoff` to `0` fixes this, but I can't quite pinpoint why this is the case (as I don't see the margin size or plot height as inputs to the automatic standoff calculation in the code. Do you have any thoughts on where this might be coming from?

Reply:
Looks to me like it's probably this block, which creates a default standoff that's not recorded anywhere or captured anywhere else, just fed into where the axis title is drawn:

https://github.com/plotly/plotly.js/blob/2b4584dfe54598db03e8d9724a766789293b4f69/src/plots/cartesian/axes.js#L3973-L3998

The ideal solution would be put all that logic into `title.standoff` in `supplyDefaults`, rather than at the draw step.

BTW @archmoj I notice `standoff` is incorrect for `multicategory` axes with `side: 'top'`... didn't look at left or right but I bet one of those is incorrect too.

Reply:
I was looking into this block as well but it's not clear to me how a change in margin size impacts the standoff value. By my reading, it's only the font size and axis properties (like tick length and line width) that we're looking at here. I could be missing something obvious since I've been staring at this block for a while.

Reply:
Look to `scootTitle` function in main `Titles.draw`